features/delete patient before import [134378119]

chrishough opened the pull request:
**Pivotal Story**
[https://www.pivotaltracker.com/story/show/134378119](https://www.pivotaltracker.com/story/show/134378119)

**Updates**

* Refactored create procedure to `delete_and_create`
* Refactored specs to verify records are deleted before created

**Specs**
```
Finished in 339.59678s
409 tests, 1973 assertions, 0 failures, 0 errors, 101 skips
Coverage report generated for Unit Tests to /Users/chrishough/Zeal/code/clients/q-centrix-health-data-standards/coverage. 6599 / 7281 LOC (90.63%) covered.
Coverage report generated for Unit Tests to /Users/chrishough/Zeal/code/clients/q-centrix-health-data-standards/coverage. 2529 / 7379 LOC (34.27%) covered.
```

chrishough committed:
delete and create method created on import process

chrishough committed:
include facility id to match specs, specs passing

chrishough committed:
adjust spec to make sure objects do not equal

chrishough committed:
remove class name since we are in the model

dillonwelch commented on lib/health-data-standards/models/record.rb:
```diff
@@ -60,15 +60,11 @@ class Record
   scope :by_provider, ->(prov, effective_date) { (effective_date) ? where(provider_queries(prov.id, effective_date)) : where('provider_performances.provider_id'=>prov.id)  }
   scope :by_patient_id, ->(id) { where(:medical_record_number => id) }
 
-  def self.update_or_create(data)
```
Can you leave the existing method and add `delete_and_create` separately? Will make future merging of updates easier.

frank-west-iii committed:
Adds back in deleted method

In order to make merging with upstream easier we want to keep the old
method and just add a new method.

frank-west-iii commented:
@oniofchaos, added back in original method

chrishough commented:
@frank-west-iii 👍
